src/ripple/nodestore/impl/Database.cpp

```diff
@@ -68,14 +74,20 @@ Database::Database(
                         std::unique_lock<std::mutex> lock(readLock_);
 
                         if (read_.empty())
+                        {
+                            runningThreads_--;
                             readCondVar_.wait(lock);
+                            runningThreads_++;
+                        }
```

Comment:
I think I see a potential deadlock. Suppose one of the threads is interrupted right after checking `while (!isStopping())`  and then another thread calls `Database::stop()`, sets `readStopping_` to true, takes the lock and calls `readCondVar_.notify_all();.`

When the thread resumes, it will wait forever on `readCondVar_.wait(lock)`, I think?

Reply:
I suggest adding :
```
if (isStopping())
    continue;
```

just after taking the lock at line 74